src/kubernetes/reflector.rs

```diff
@@ -82,15 +82,11 @@ where
             let invocation_result = self.issue_request().await;
             let stream = match invocation_result {
                 Ok(val) => val,
-                Err(watcher::invocation::Error::Desync { source }) => {
-                    emit!(internal_events::InvocationDesyncReceived { error: source });
-                    // We got desynced, reset the state and retry fetching.
-                    self.resource_version.reset();
-                    self.state_writer.resync().await;
+                Err(watcher::invocation::Error::Recoverable { source }) => {
+                    emit!(internal_events::InvocationHttpErrorReceived { error: source });
                     continue;
                 }
                 Err(watcher::invocation::Error::Other { source }) => {
-                    // Not a desync, fail everything.
                     error!(message = "Watcher error.", error = ?source);
                     return Err(Error::Invocation { source });
```

Comment:
I think we might actually just want to retry _any_ error we get rather than specifically only retrying HTTP errors. Encountering an error that hits this match arm is going to result in the same behavior described by https://github.com/timberio/vector/issues/6795 where the `kubernetes_logs` component just stops processing but Vector continues to run.

Reply:
Not here, this should be done at the watcher level, and the abstraction at this level shouldn't deal with transport errors at all.